Snowflake.Client/SnowflakeClient.cs

```diff
@@ -117,14 +116,13 @@ public async Task<bool> RenewSessionAsync()
             if (_session == null)
                 throw new SnowflakeException("Session is not itialized yet.");
 
-            var renewSessionRequest = _requestBuilder.BuildRenewSessionRequest();
+            var renewSessionRequest = _requestBuilder.BuildRenewSessionRequest(_session);
             var response = await _restClient.SendAsync<RenewSessionResponse>(renewSessionRequest).ConfigureAwait(false);
 
             if (!response.Success)
                 throw new SnowflakeException($"Renew session failed. Message: {response.Message}", response.Code);
 
-            _session.Renew(response.Data);
-            _requestBuilder.SetSessionTokens(_session.SessionToken, _session.MasterToken);
```

Comment:
Maybe redo `SetSessionTokens` as `SetSession`? Then we don't have to pass the session into all the methods.

Reply:
I had this in an earlier version, something like:
```csharp
_session = new SnowflakeSession(...);
_requestBuilder.SetSession(_session);
```
but was concerned about lock-free thread-safety here too, particularly as you could transiently get `_session != null` before you've poked the session/tokens into `_requestBuilder`.

Then I played with
```csharp
var session = new SnowflakeSession(...);
_requestBuilder.SetSession(session);
_session = session;
```
but thought it looked ugly.